docker/daemon.go

```diff
@@ -141,7 +153,17 @@ func mainDaemon() {
 	job.Setenv("TlsCert", *flCert)
 	job.Setenv("TlsKey", *flKey)
 	job.SetenvBool("BufferRequests", true)
-	if err := job.Run(); err != nil {
-		log.Fatal(err)
+	err = job.Run()
+
+	// Wait for the daemon startup goroutine to finish
+	// This makes sure we can actually cleanly shutdown the daemon
+	wg.Wait()
+	eng.Shutdown()
+	if err != nil {
+		// log errors here so the log output looks more consistent
+		log.Error(err)
+		log.Error("shutting down daemon due to errors")
```

Comment:
might as well do `log.Fatalf("shutting down daemon due to errors %v", err)` for this one :)